protocol/messenger_group_chat_test.go

```diff
@@ -264,6 +264,12 @@ func (s *MessengerGroupChatSuite) TestGroupChatMembersRemoval() {
 	s.makeMutualContacts(admin, memberA)
 	s.makeMutualContacts(admin, memberB)
 	s.makeMutualContacts(admin, memberC)
+	// if we expect r.Chats()[0].Active == true when #verifyGroupChatCreated
+	// then we need make admin/memberA/memberB/memberC as mutual contacts, not only just with admin
+	// since memberB may receive protobuf.MembershipUpdateMessage from memberA rather than admin
+	s.makeMutualContacts(memberA, memberB)
+	s.makeMutualContacts(memberA, memberC)
+	s.makeMutualContacts(memberB, memberC)
```

Comment:
I do not understand why they should be mutual. We can have some user create a group, then add 3 other users from his contacts who do not know each other to the group, right?

Reply:
> I do not understand why they should be mutual. We can have some user create a group, then add 3 other users from his contacts who do not know each other to the group, right?

right, but then we got inactive chat

Reply:
Alright, LGTM but it's better if someone else looks at this also. So what would actually happen if someone creates a group then adds some users who are mutual with each other, then leaves the group?

Reply:
i'm trying to find a better solution to fix this test, since we have 
```golang
s.makeMutualContacts(admin, memberA)
s.makeMutualContacts(admin, memberB)
s.makeMutualContacts(admin, memberC)
```
and admin sent  protobuf.MembershipUpdateMessage to each member already.
We can't ensure the order of message received, but we should figure out a better way to handle this situation

Reply:
Thanks @qfrank for approaching this problem.

> We can't ensure the order of message received, but we should figure out a better way to handle this situation

I agree. The solution of making all members mutual is just masking the issue. Also, it makes the intention of the test unclear, as mentioned [here](https://github.com/status-im/status-go/pull/3037#discussion_r1057481203).

The problem should be solved in `verifyGroupChatCreated`. It should handle "out of order messages issue". Theoretically, it should work fine just now, because it uses `ExponentialBackOff` behind the scenes, meaning it will retry the operation until it does not return an error. In other words, it will wait for the response where the given condition is true (and it should be true when receiver eventually gets the message from the admin).
```
	response, err := WaitOnMessengerResponse(
		member,
		func(r *MessengerResponse) bool {
			return len(r.Chats()) == 1 && r.Chats()[0].Active == expectedChatActive
		},
		"chat invitation not received",
	)
```

The root cause is either:
a) `ExponentialBackOff` failure, we can verify that by inspecting if the receiver got the message from the admin
b) bug in group chat logic, when we first get the `MembershipUpdateMessage` from member rather than admin, resulting in chat.Active == false

Reply:
I discussed this issue with Andrea before, because I discovered the same thing when investigating this desktop issue https://github.com/status-im/status-desktop/issues/8773 (you might be able to see the discussion in the status-go channel, it's from Decmber 16th)

Basically, this is the way to fix the test for now. 
Like you rightly pointed out @qfrank , if the membership update comes from a non-mutual contact, that channel will be made inactive.
That is because we wanted to negate the spam. Back in the day, group chats could be created by anyone and with anyone, so you'd get spammed with group chats and you had to manually reject them. Now, you don't receive them from non-mutual contacts to negate that. 
It does create the annoying scenario above when if the person that adds you, or like in the issue I linked, if you re-import your account and the signal comes in from a non-mutual contact (from a normal chat message for example), then the chat will be marked as inactive.

The solution we'll do in desktop is show that invite in the AC only. I think mobile does something similar already. We're just waiting for designs.

I hope it was clear and makes sense.
One way to improve this test further would be to voluntarily test the negative case where if the signal comes from a non-mutual contact, then we assert that `Active` will be `false`.